### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by SamirPS**
## Description

I create a makefile to simplify answer this issue : https://github.com/aphp/edsnlp/issues/6 #6 

Now you just need to do this command
## Checklist
make create-env (to create the virtual env) 
make install  (install the application) 
make documentation (build and launch the documentation) 
make test (launch the test) 

- [ ] If this PR is a bug fix, the bug is documented in the test suite.
- [ ] Changes were documented in the changelog (pending section).
- [ ] If necessary, changes were made to the documentation (eg new pipeline).

### CODE REVIEW

**Comment by SamirPS** [comment]
@bdura

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+
+.ONESHELL:
+	SHELL:=/bin/bash 
+
+.PHONY: create-env install documentation test
+
+default:
+	@echo "Call a specific subcommand: create-env,install,documentation,test"
+
+create-env:
```
Comment (percevalw) [inline Makefile:10]: I would suggest to change the target name to the venv directory name (`.venv`) here to enable make dependency checking features, or to create another non-phony target
```
.venv:
   ...
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	. venv/bin/activate
+	pip install -r requirements.txt
+	pip install -r requirements-setup.txt
+	python scripts/conjugate_verbs.py
+	pip install -r requirements-dev.txt
+	pip install -r requirements-docs.txt
+	pip install -e .
+	pre-commit install
+
+documentation:
```
Comment (percevalw) [inline Makefile:24]: Same here
```suggestion
documentation: .venv
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+default:
+	@echo "Call a specific subcommand: create-env,install,documentation,test"
+
+create-env:
+	
+	python -m venv venv
+
+install : 
+	. venv/bin/activate
```
Comment (percevalw) [inline Makefile:15]:
```suggestion
	. .venv/bin/activate
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	SHELL:=/bin/bash 
+
+.PHONY: create-env install documentation test
+
+default:
+	@echo "Call a specific subcommand: create-env,install,documentation,test"
+
+create-env:
+	
+	python -m venv venv
```
[the remark this answers is not in the document]
Comment (percevalw) [reply Makefile:12]: We could also make this name a variable

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	pip install -r requirements-docs.txt
+	pip install -e .
+	pre-commit install
+
+documentation: 
+	. venv/bin/activate
+	pip install -r requirements-docs.txt
+	mkdocs serve
+
+test:
```
Comment (percevalw) [inline Makefile:29]: Same here
```suggestion
test: .venv
```

**Review by percevalw — COMMENTED** [review]
Hi Samir, thanks for this contribution! I'm not sure we absolutely need a Makefile but I'm not against it either :) I have made some comments and suggestions

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (SamirPS) [reply Makefile]: I update the name of the venv now it's  python -m venv .venv

File: Makefile (hunk 1 above)
Comment (SamirPS) [reply Makefile]: it's change

**Comment by SamirPS** [comment]
I commit a new Makefile with your suggestion, tell me if you find other things       @percevalw

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+	SHELL:=/bin/bash 
+
+.PHONY: .venv install documentation test
+
+default:
+	@echo "Call a specific subcommand: .venv,install,documentation,test"
+
+.venv:
+	
+	python3 -m venv .venv
```
Comment (percevalw) [inline Makefile:12]:
```suggestion
.venv:
	python3 -m venv .venv
	
create-env: .venv
```

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,31 @@
+
+.ONESHELL:
+	SHELL:=/bin/bash 
+
+.PHONY: .venv install documentation test
+
+default:
+	@echo "Call a specific subcommand: .venv,install,documentation,test"
```
Comment (percevalw) [inline Makefile:8]:
```suggestion
	@echo "Call a specific subcommand: create-env,install,documentation,test"
```

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,31 @@
+
+.ONESHELL:
+	SHELL:=/bin/bash 
+
+.PHONY: .venv install documentation test
```
Comment (percevalw) [inline Makefile:5]:
```suggestion
.PHONY: create-env install documentation test
```

**Review by percevalw — COMMENTED** [review]
Thanks ! `create-env` was useful, we could keep it but have a `.venv` non-phony target.

Could you also please edit the `CHANGELOG.md` file to mention this PR ? Thank you, almost there :)

**Comment by SamirPS** [comment]
Normally it's good now @percevalw

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+	pre-commit install
+
+documentation: .venv
+	. .venv/bin/activate
+	pip install -r requirements-docs.txt
+	mkdocs serve
+
+test: .venv
+	. .venv/bin/activate
+	python3 -m pytest
```
Comment (percevalw) [inline Makefile:32]: For the linter !

```suggestion
	python3 -m pytest

```

**Comment by SamirPS** [comment]
Good,i also change python3 to python because in most cast python refer to python 3

**Comment by SamirPS** [comment]
@percevalw i see that linter fail, but i don't see why in the logs

**Comment by percevalw** [comment]
Did you run `pre-commit run --all-files` or `pre-commit install` before commiting ?

**Comment by SamirPS** [comment]
No sorry, when i can I will do it

**Comment by percevalw** [comment]
See our [contribution guidelines](https://github.com/aphp/edsnlp/blob/master/contributing.md)

**Comment by SamirPS** [comment]
Normally now it's good @percevalw

**Review by percevalw — APPROVED** [review]
Great thanks !

**Comment by SamirPS** [comment]
It's nothing thanks to you

**Pull request merged**